### PR TITLE
Cheat n times docs

### PR DESCRIPTION
**Pull request by drknzz**

### CODE REVIEW

File: docs/src/appendix/cheatcodes/caller_address/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# caller_address
+
+Cheatcodes modifying caller_address:
+
+* [`prank`](./prank.md) - changes the caller address for the given target, for given span
+* [`start_prank`](./start_prank.md) - changes the caller address for the given target
```
[the remark this answers is not in the document]
Comment (Arcticae) [reply docs/src/appendix/cheatcodes/caller_address/README.md:6]: "until stop is called"

**Review by Arcticae — APPROVED** [review]

File: docs/src/appendix/cheatcodes/caller_address/start_prank.md
[the remark this answers is not in the document]
Comment (drknzz) [reply docs/src/appendix/cheatcodes/caller_address/start_prank.md:1]: `prank.md` doesn't have any examples, only the signature + explanation
either way, it should hopefully be automated soon

File: docs/src/SUMMARY.md [hunk 2]
```diff
@@ -46,21 +46,23 @@
     * [clean-cache](appendix/snforge/clean-cache.md)
 * [Cheatcodes Reference](appendix/cheatcodes.md)
     * [`CheatTarget`](appendix/cheatcodes/cheat_target.md)
-    * [start_prank](appendix/cheatcodes/start_prank.md)
-    * [stop_prank](appendix/cheatcodes/stop_prank.md)
+    * [`CheatSpan`](appendix/cheatcodes/cheat_span.md)
+    * [caller_address](appendix/cheatcodes/caller_address/README.md)
```
[the remark this answers is not in the document]
Comment (Arcticae) [reply docs/src/SUMMARY.md:50]: I'd leave as-is

File: docs/src/appendix/cheatcodes.md [hunk 3]
```diff
@@ -1,19 +1,27 @@
...
+- [`stop_warp`](cheatcodes/block_timestamp/stop_warp.md) - cancels the `start_warp` for contracts
+- [`elect`](cheatcodes/sequencer_address/elect.md) - changes the sequencer address for contracts, for a number of calls
+- [`start_elect`](cheatcodes/sequencer_address/start_elect.md) - changes the sequencer address for contracts
+- [`stop_elect`](cheatcodes/sequencer_address/stop_elect.md) - cancels the `start_elect` for contracts
+- [`spoof`](cheatcodes/tx_info/spoof.md) - changes the transaction context for contracts, for a number of calls
+- [`start_spoof`](cheatcodes/tx_info/start_spoof.md) - changes the transaction context for contracts
+- [`stop_spoof`](cheatcodes/tx_info/stop_spoof.md) - cancels the `start_spoof` for contracts
+- [`mock_call`](cheatcodes/mock/mock_call.md) - mocks contract call to an entry point, for a number of calls
+- [`start_mock_call`](cheatcodes/mock/start_mock_call.md) - mocks contract call to an entry point
+- [`stop_mock_call`](cheatcodes/mock/stop_mock_call.md) - cancels the `start_mock_call` for an entry point
```
Comment (Arcticae) [inline docs/src/appendix/cheatcodes.md:22]: doesn't it cancel regular spoofs? (applies for all above)

File: docs/src/appendix/cheatcodes/cheat_span.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+```rust
+enum CheatSpan {
+    Indefinite: (),
+    Calls: usize
+}
+```
+
+`CheatSpan` is an enum used to specify for how long the target should be cheated.
+- `Indefinite` applies the cheatcode for indefinitely, until the cheat is stopped manually (e.g. using `stop_warp`).
```
Comment (Arcticae) [inline docs/src/appendix/cheatcodes/cheat_span.md:11]:
```suggestion
- `Indefinite` applies the cheatcode indefinitely, until the cheat is stopped manually (e.g. using `stop_warp`).
```

File: docs/src/appendix/cheatcodes/cheat_span.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+- `Calls` applies the cheatcode for specified number of calls, after which the cheat is stopped.
```
Comment (Arcticae) [inline docs/src/appendix/cheatcodes/cheat_span.md:12]: I think we should call this behavior consistently. 
One time it appears as "cancelled" one time it's "stopped"

File: docs/src/testing/using-cheatcodes.md [hunk 6]
```diff
@@ -226,3 +226,72 @@ fn mock_constructor_with_prank() {
...
+}
+```
+
+To set span for a cheatcode, use `prank` / `warp` / `roll` / etc.
+
+```rust
+prank(CheatTarget::One(contract_address), new_caller_address, CheatSpan::Calls(1))
+```
+
+Calling a cheatcode with `CheatSpan::Calls(N)` is going to activate the cheatcode for `N` calls to a contract, after which it's going to be automatically cancelled.
```
Comment (Arcticae) [inline docs/src/testing/using-cheatcodes.md:249]:
```suggestion
Calling a cheatcode with `CheatSpan::Calls(N)` is going to activate the cheatcode for `N` calls to a contract, after which it's going to be automatically canceled.
```

Or the other way around in other places.

File: docs/src/appendix/cheatcodes/caller_address/prank.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# `prank`
+
+> `fn prank(target: CheatTarget, caller_address: ContractAddress, span: CheatSpan)`
+
+Changes the caller address for the given target and span.
+This change can be canceled with [`stop_prank`](./stop_prank.md).
+
+- `target` - instance of [`CheatTarget`](../cheat_target.md) specifying which contracts to prank
+- `caller_address` - caller address to be set
+- `span` - instance of [`CheatSpan`](../cheat_span.md) specifying the span of prank
```
Comment (MaksymilianDemitraszek) [inline docs/src/appendix/cheatcodes/caller_address/prank.md:10]: xD

**Review by MaksymilianDemitraszek — APPROVED** [review]

File: docs/src/appendix/cheatcodes/caller_address/prank.md (hunk 7 above)
Comment (drknzz) [reply docs/src/appendix/cheatcodes/caller_address/prank.md:10]: You suggested not to use duration, so :D

Comment (MaksymilianDemitraszek) [reply docs/src/appendix/cheatcodes/caller_address/prank.md:10]: What I meant is: if you are using duration to explain span, maybe duration is a better name after all :D . I think what we have now is ok.

**Pull request merged**